# About the Project

## Introduction
Standardized testing, particularly the SATs, holds significant weight in the American education system, serving as a crucial determinant for college admissions. With each of its three sections—reading, math, and writing—carrying a maximum score of 800 points, the SATs play a pivotal role in assessing students' literacy, numeracy, and writing skills. Analyzing the performance of schools on these tests is paramount for a multitude of stakeholders, including policymakers, educators, researchers, governmental bodies, and parents deliberating on their children's educational paths.

## Dataset Overview
For this project, we have been provided with a dataset named schools.csv, which encapsulates valuable insights into New York City (NYC) public school SAT performance. This dataset serves as the foundation for our analysis, containing crucial metrics related to SAT scores and other relevant factors. By delving into this dataset, we aim to unravel patterns, trends, and performance metrics of NYC public schools in SAT examinations.

## Key Questions

Our analysis revolves around three central questions concerning NYC public school SAT performance:

1. Which NYC schools have the best math results?
2. What are the top 10 performing schools based on the combined SAT scores?
3. Which single borough has the largest standard deviation in the combined SAT score?

Through rigorous examination and data-driven insights, we endeavor to shed light on these inquiries, providing valuable insights for stakeholders invested in understanding and improving NYC public school education.





## Question 1: Which NYC schools have the best math results?

To determine the schools with the best math results, we establish a threshold of at least 80% of the maximum possible score of 800 for math. This criterion ensures that we focus on schools where students perform exceptionally well in mathematics.

In [38]:
# Re-run this cell 
import pandas as pd
from tabulate import tabulate

# Read in the data
schools = pd.read_csv(r'H:\Project\Exploring_NYC_Public_School_Test_Result_Scores\Data Source\schools.csv')

# Preview the data
schools.head()

# NYC schools that have the best math results
filtered_schools = schools[schools['average_math'] / 800 >= 0.8]
best_math_schools = filtered_schools[['school_name', 'average_math']].sort_values(by='average_math', ascending=False)
best_math_schools=best_math_schools.reset_index(drop=True)
best_math_schools_table = tabulate(best_math_schools, headers='keys', tablefmt='pipe')
print(best_math_schools_table)


|    | school_name                                                           |   average_math |
|---:|:----------------------------------------------------------------------|---------------:|
|  0 | Stuyvesant High School                                                |            754 |
|  1 | Bronx High School of Science                                          |            714 |
|  2 | Staten Island Technical High School                                   |            711 |
|  3 | Queens High School for the Sciences at York College                   |            701 |
|  4 | High School for Mathematics, Science, and Engineering at City College |            683 |
|  5 | Brooklyn Technical High School                                        |            682 |
|  6 | Townsend Harris High School                                           |            680 |
|  7 | High School of American Studies at Lehman College                     |            669 |
|  8 | New Explorations into Science, Te

### Best Math Schools: NYC's Top Performers in Math

**Stuyvesant High School** leads in math performance among NYC schools, with an average math score of 754, followed closely by **Bronx High School of Science** and **Staten Island Technical High School**, reflecting excellence in math education.

## Question 2: What are the top 10 performing schools based on the combined SAT scores?

To identify the top-performing schools based on the combined SAT scores, we aggregate the scores from the reading, math, and writing sections for each school and rank them accordingly. The resulting DataFrame, named top_10_schools, contains the school names along with their combined SAT scores, sorted in descending order of total SAT scores.

In [39]:
# The top 10 performing schools based on the combined SAT scores
schools['total_SAT']=schools['average_math']+schools['average_reading']+schools['average_writing']
top_10_schools=schools[['school_name', 'total_SAT']].sort_values(by='total_SAT', ascending=False)
top_10_schools = top_10_schools.head(10)
top_10_schools = top_10_schools.reset_index(drop=True)
top_10_schools_table = tabulate(top_10_schools, headers='keys', tablefmt='pipe')
print(top_10_schools_table)

|    | school_name                                                           |   total_SAT |
|---:|:----------------------------------------------------------------------|------------:|
|  0 | Stuyvesant High School                                                |        2144 |
|  1 | Bronx High School of Science                                          |        2041 |
|  2 | Staten Island Technical High School                                   |        2041 |
|  3 | High School of American Studies at Lehman College                     |        2013 |
|  4 | Townsend Harris High School                                           |        1981 |
|  5 | Queens High School for the Sciences at York College                   |        1947 |
|  6 | Bard High School Early College                                        |        1914 |
|  7 | Brooklyn Technical High School                                        |        1896 |
|  8 | Eleanor Roosevelt High School                                  

### Best NYC Schools: Topping the Charts in SAT Scores

**Stuyvesant High School** leads the pack with an impressive combined SAT score of 2144, followed closely by **Bronx High School of Science** and **Staten Island Technical High School**, showcasing excellence in SAT performance among NYC schools.

## Question 3: Which single borough has the largest standard deviation in the combined SAT score?

To determine the NYC borough with the largest standard deviation in the combined SAT score, we compute the standard deviation of the combined SAT scores for each borough. The resulting DataFrame, named largest_std_dev, provides insights into the variability of SAT scores across different boroughs.

In [40]:
# The largest standard deviation in the combined SAT score of the single borough

boroughs = schools.groupby("borough")["total_SAT"].agg(["count", "mean", "std"]).round(2)

# Filter for max std and make borough a column
largest_std_dev = boroughs[boroughs["std"] == boroughs["std"].max()]

# Rename the columns for clarity
largest_std_dev = largest_std_dev.rename(columns={"count": "num_schools", "mean": "average_SAT", "std": "std_SAT"})
largest_std_dev_table = tabulate(largest_std_dev, headers='keys', tablefmt='pipe')
print(largest_std_dev_table)

| borough   |   num_schools |   average_SAT |   std_SAT |
|:----------|--------------:|--------------:|----------:|
| Manhattan |            89 |       1340.13 |    230.29 |


### Manhattan Schools Showcase Diverse SAT Performance: A Tale of Variability

The borough of **Manhattan** exhibits a notable variability in SAT scores among its 89 schools, with an average combined SAT score of 1340.13 and a standard deviation of 230.29, suggesting diverse academic performance levels across schools in the borough.